src/DotNet.Testcontainers/Containers/Configurations/Databases/CouchbaseTestcontainerConfiguration.cs

```diff
@@ -0,0 +1,101 @@
+namespace DotNet.Testcontainers.Containers.Configurations.Databases
+{
+  using System;
+  using System.IO;
+  using Abstractions;
+  using OutputConsumers;
+  using WaitStrategies;
+
+  public sealed class CouchbaseTestcontainerConfiguration : TestcontainerDatabaseConfiguration
+  {
+    internal const string CouchbaseImage = "mustafaonuraydin/couchbase-testcontainer:6.0.0";
+    private const string WaitStrategyMessage = "couchbase-dev started";
+
+    private const string DefaultIndexStorage = "memopt";
+
+    private const int DefaultClusterRamSize = 1024;
+    private const int DefaultClusterIndexRamSize = 512;
+    private const int DefaultClusterEventingRamSize = 256;
+    private const int DefaultClusterFtsRamSize = 256;
+    private const int DefaultClusterAnalyticsRamSize = 1024;
+
+    private const int BootstrapHttpPort = 8091;
+
+    private Stream OutStream = new MemoryStream();
+    private Stream ErrorStream = new MemoryStream();
+
+
+    public override IOutputConsumer OutputConsumer => Consume.RedirectStdoutAndStderrToStream(this.OutStream, this.ErrorStream);
+
+    public override IWaitForContainerOS WaitStrategy => Wait.ForUnixContainer().UntilMessageIsLogged(this.OutputConsumer.Stdout, WaitStrategyMessage);
+
+    public CouchbaseTestcontainerConfiguration(string username, string password, string bucket = "Sample", string bucketType = "couchbase", int bucketRamSize = 512)
+      : base(CouchbaseImage, BootstrapHttpPort, BootstrapHttpPort)
+    {
+      this.Username = username;
+      this.Password = password;
+      this.Port = BootstrapHttpPort;
+      this.Environments["ADMIN_NAME"] = username;
+      this.Environments["PASSWORD"] = password;
+      this.Environments["INITIAL_BUCKET_NAME"] = bucket;
+      this.Environments["INITIAL_BUCKET_PASSWORD"] = password;
+      this.Environments["INITIAL_BUCKET_TYPE"] = bucketType;
+      this.Environments["INITIAL_BUCKET_RAMSIZE"] = bucketRamSize.ToString();
+      this.InitializeDefaults();
```

Comment:
Can we call an empty constructor instead?

Reply:
We can remove that initializing after adding docker image sources into project. Default variables will be already set to defaults.